src/when.js

```diff
@@ -0,0 +1,34 @@
+var _curry2 = require('./internal/_curry2');
+var _identity = require('./internal/_identity');
+var ifElse = require('./ifElse');
+
+
+/**
+ * Creates a function that will first pass its argument to the given predicate
+ * function. If the predicate returns true, the function will return the result
```

Comment:
It needn't actually return the `true` value. Perhaps we could write _if the predicate is satisfied_ to sidestep the matter of implicit type coercion.